iris-mpc-gpu/src/server/actor.rs

```diff
@@ -1083,6 +1088,7 @@ impl ServerActor {
                 store_right: query_store_right,
                 deleted_ids: batch.deletion_requests_indices,
                 matched_batch_request_ids,
+                anonymized_bucket_statistics: self.anonymized_bucket_statistics.clone(),
```

Comment:
since this is never emptied, is this not sent on every response once the buckets have been calculated?
Maybe having `self.anonymized_bucket_statistics` being an option instead and `take`ing it here if full could be simpler?

Reply:
The inner property `buckets` content is emptied: https://github.com/worldcoin/iris-mpc/pull/989/files#diff-cfedd06c9bac2141ada3a2a2664595d46a77fbb2811d94c2c0a3a15302141cb0R66 

Though I am working in a fix to set the correct start and end timestamp, as currently they are not exactly what we want. I think making it an option won't work, because we'll never be able to set the start_timestamp (unless we store it as ServerActor property)

Reply:
yeah but if it is only ever emptied in fill_buckets, then will it not be sent until it is cleared (and then subsequently filled again immediately) in the next call to fill_buckets (which only happens every 2^19 matches atm)?

I think you also need to clear it after a successful send to SNS, unless I am misunderstanding something?

Reply:
you definitely have a point there